src/systemctl/systemctl-show.c

```diff
@@ -1889,13 +1891,32 @@ static int show_one(
         if (r < 0)
                 return log_error_errno(r, "Failed to rewind: %s", bus_error_message(&error, r));
 
-        r = bus_message_print_all_properties(reply, print_property, arg_properties, arg_value, arg_all, &found_properties);
-        if (r < 0)
-                return bus_log_parse_error(r);
+        if (!FLAGS_SET(arg_json_format_flags, JSON_FORMAT_OFF)) {
```

Comment:
Why not revert the codnition (and the order of branches)? Positive conditions are easier to think about...